src/lib/message.ts

```diff
@@ -43,6 +47,7 @@ class GleeMessage extends EventEmitter {
    * @param {Object} [options.headers] Message headers.
    * @param {String} [options.channel] Message channel.
    * @param {String} [options.serverName] The name of the associated AsyncAPI server.
+   * @param {GleeMessage} [options.request] The request that this message is created for as a reply.
```

Comment:
> The request that this message is created for as a reply.

What? 😅  Would you mind to elaborate?

Reply:
In the previouse codebase we were manipulating the message to create a reply. I suggest we create an entirely new message and keep a reference to the parent(request) messge here.
here is a few benifits:
1) by not modefying the message we keep the codebase simpler.
2) The hierarcy of the request/reply will be kept. for example here: 
https://github.com/asyncapi/glee/blob/8e5a1a148927e4fcff07b166285eb4dc60f1e264/examples/slack-reaction-listener/functions/ReceiveFromOpenAI.ts#L4

I am able to trace the request that the reply message was created for.

not sure if this is the best way of doing it though.